skunkworks_crow/src/main/java/org/odk/share/views/ui/settings/CustomEditTextPreference.java

```diff
@@ -0,0 +1,73 @@
+package org.odk.share.views.ui.settings;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.preference.EditTextPreference;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.widget.Button;
+
+/**
+ * @author by Chromicle (ajayprabhakar369@gmail.com)
+ * @since 1/30/2020
+ */
+public class CustomEditTextPreference extends EditTextPreference implements DialogInterface.OnClickListener {
```

Comment:
I'm not sure do we really need a custom preference for this tiny check? @Chromicle 

Can we use some method like,

```java
private void checkNullEditTextPreference(EditTextPreference edttxtpref) {
    edttxtpref.setOnPreferenceChangeListener(new OnPreferenceChangeListener() {
        @Override
        public boolean onPreferenceChange(
                android.preference.Preference preference, Object newValue) {
            if (newValue.toString().trim().equals("")) {

              // Get Dialog and disable button here

                return false;
            }
            return true;
        }
    });
}
```
and receiving a EditTextPreference? You know, building a custom class is always a heavy work and not good for code reading and code refactoring. I didn't test for this method but you can have a try.

Reply:
As far as I know, `setOnPreferenceChangeListener` will be triggered when we press the positive button of the dialog so there is no use if we disable the button then so, here we want to check the edit text for every time so we have to implement the `editText watcher`  correct me if I am wrong 

Now I updated the code without implementing the `customEditTextPreference` can please have a look now and thanks for your suggestions 😄